proof_of_stake/src/parameters.rs

```diff
@@ -113,6 +113,10 @@ impl PosParams {
 
         // Check maximum total voting power cannot get larger than what
         // Tendermint allows
+        //
+        // TODO: decide if this is still a check we want to do (in its current
+        // state with our latest voting power conventions, it will fail
+        // always)
```

Comment:
I actually think I see a bug in here rn... `self.tm_votes_per_token` is relative to namnam, but `TOKEN_MAX_AMOUNT` is in terms of NAM. I think my point was that I don't really see the point of this check here, since I think the appropriate parameter to consider would be some upper bound on the total allowed supply of NAM (which of course is not actually specified or built in). Because right now with `self.tm_votes_per_token` relative to 1 namnam and validator amounts stored as `u64`, this check will always fail since the max tendermint voting power is `i64::MAX/8`.